daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/HexBinaryLengthParsers.scala

```diff
@@ -36,6 +36,9 @@ sealed abstract class HexBinaryLengthParser(override val context: ElementRuntime
     val nBits = getLengthInBits(start).toInt
     if (nBits == 0) {
       currentElement.setDataValue(zeroLengthArray)
+    } else if (nBits < 0) {
+      // Integer overflow occurred, recommend using blob instead of hexBinary
```

Comment:
We're depending on 32-bit integer overflow? That's very unreliable. We should do math on Long type, and check the bound, and explicitly return -1 to indicate too big. 

Or, if we are doing this right and signaling the overflow of the max length by -1 then the comment should change.

Reply:
This has been addressed in my most recent commit